Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Sync/DataSynchronizer.swift

```diff
@@ -1621,26 +1613,28 @@ public class DataSynchronizer: NetworkStateDelegate, FatalErrorListener {
         var deferredEventEmittingBlock: DeferredBlock? = nil
```

Comment:
[nit] with this new locking scheme, we don't need the deferred event emitting block. Since we're not deferring to unlock the lock, we can just trigger listening and emit the event at the end of the function outside of the `nsLock.write` block

Reply:
My comments about getting rid of the deferred event emitting block are still unaddressed. We can leave as is if we think we might move away from this locking mechanism eventually but if we're sticking with the dispatch approach then you can move anything inside the deferred event emitting block at the end to just occur outside the locked block